src/bptree/bptree.ts

```diff
@@ -78,22 +88,28 @@ export class BPTree {
 			return null;
 		}
 
-		const path = await this.traverse(key, child);
+		const path = await this.traverse(
+			key,
+			child,
+			node.pointers[node.pointers.length - 1]
+		);
 		if (!path) {
 			return null;
 		}
 
-		return [...path, { node: node, index: node.keys.length }];
+		return [...path, { node: node, index: node.keys.length, pointer: pointer }];
```

Comment:
The traversal refactor was to update the `TraversalRecord` to include a memory pointer.

Am I missing something or is there more to refactor? @kevmo314

Reply:
Hmm, nominally the logic looks correct. Is this able to read the index file generated by the btree version?

Reply:
I will get everything connected 👍